core/pkg/server/handler.go

```diff
@@ -351,7 +351,12 @@ func (h *Handler) handleRequestLogin(record *spb.Record) {
 }
 
 func (h *Handler) handleRequestCheckVersion(record *spb.Record) {
-	h.fwdRecord(record)
+	if h.settings.GetXOffline().GetValue() {
+		// Send an empty response if we're offline.
+		h.respond(record, &spb.Response{})
+	} else {
+		h.fwdRecord(record)
+	}
```

Comment:
wait maybe it should be respond and fwd (as the writer won't send it further, since it is a request?)

Reply:
should be equivalent! i'm trying to remove this stuff soon anyway---it would be neater to do `fwdRecord` by default, with explicit exceptions in cases where we don't want to

Reply:
yeah it would, ah that would be nice agreed

Reply:
also i think we should modify the respond function a bit to check for the controls (maybe even move it the responder struct?)

Reply:
i agree, let me look into that. I will send separate PRs